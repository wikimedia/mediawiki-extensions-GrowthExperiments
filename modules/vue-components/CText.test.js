const { mount } = require( '@vue/test-utils' );
const CText = require( './CText.vue' );

// TODO assert for all render modes
const renderComponent = ( { props, slots = { default: 'Some text' }, RENDER_MODE = 'desktop' } ) => {
	return mount( CText, {
		props,
		slots,
		global: {
			provide: {
				RENDER_MODE
			},
			mocks: {
				$filters: {
					convertNumber: jest.fn( ( x ) => `localised-${x}` )
				}
			}
		}
	} );
};

const TEST_INPUTS = [
	{
		props: {}
	},
	{
		props: {
			color: 'subtle',
			size: 'l',
			weight: 'bold'
		}
	},
	{
		props: {
			size: [ 'xl', null, null ]
		}
	},
	{
		props: {},
		slots: {
			default: '<a href="#">Some link</a>'
		}
	}
];

describe( 'CText', () => {
	it( 'should render correctly', () => {
		for ( const props of TEST_INPUTS ) {
			const wrapper = renderComponent( props );
			expect( wrapper.element ).toMatchSnapshot();
		}
	} );
} );
