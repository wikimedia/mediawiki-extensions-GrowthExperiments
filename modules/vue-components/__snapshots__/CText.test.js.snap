// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CText should render correctly 1`] = `
<div
  class="ext-growthExperiments-CText"
>
  
  Some text
  
</div>
`;

exports[`CText should render correctly 2`] = `
<div
  class="ext-growthExperiments-CText ext-growthExperiments-CText--size-l ext-growthExperiments-CText--color-subtle ext-growthExperiments-CText--weight-bold"
>
  
  Some text
  
</div>
`;

exports[`CText should render correctly 3`] = `
<div
  class="ext-growthExperiments-CText ext-growthExperiments-CText--size-xl"
>
  
  Some text
  
</div>
`;

exports[`CText should render correctly 4`] = `
<div
  class="ext-growthExperiments-CText"
>
  
  <a
    href="#"
  >
    Some link
  </a>
  
</div>
`;
